Tests/SentryTests/PrivateSentrySDKOnlyTests.swift

```diff
@@ -29,6 +56,18 @@ class PrivateSentrySDKOnlyTests: XCTestCase {
         XCTAssertEqual(1, client?.captureEnvelopeInvocations.count)
         XCTAssertEqual(envelope, client?.captureEnvelopeInvocations.first)
     }
+    
+    func testcaptureEnvelopeWithUndhandledMarksSessionAsCrashedAndStartsNewSession() {
+        let client = TestClient(options: Options())
+        let hub = TestHub(client: client, andScope: nil)
+        SentrySDK.setCurrentHub(hub)
+        hub.setTestSession()
+
+        let envelope = Fixture.getUnhandledExceptionEnvelope()
+        PrivateSentrySDKOnly.capture(envelope)
+
+        XCTAssertEqual(1, hub.startSessionInvocations)
```

Comment:
`m`: This test doesn't validate that the SDK marks the session as crashed. It would be great if it did.